webapp/channels/src/plugins/export.js

```diff
@@ -103,3 +104,6 @@ window.ProductApi = {
     getRhsSelectedPostId: getSelectedPostId,
     getIsRhsOpen,
 };
+
+// Desktop App module containing the app info and a series of helpers to work with legacy code
+window.desktopApp = DesktopApp;
```

Comment:
How are we expecting plugins to interact with the desktop API? I would've expected the web app to expose some compatibility-friendly version of the desktop API to them instead of them just getting it from `window.desktopAPI`.

If we don't make the compatibility-friendly versions of those methods part of the `DesktopApp` like I was thinking about elsewhere, do we actually need to expose the `DesktopApp` itself to plugins? At the moment, it looks like it mostly contains setup and glue code that they wouldn't need direct access to

Reply:
So the `DesktopApp` class contains the functionality to register legacy listeners, and it also has the name/version. I wanted to make sure plugins had access to that at a minimum.

I was interested in sharing the other methods as well, but as it stands I don't think any plugins use it currently. But I would like to keep the API consistent and belonging to the Web App so we can share it with the plugins.

Reply:
I agree that the web app should access the methods in the same way as a plugin would, but I think that should be via `DesktopApp` or a different class as mentioned above

Reply:
Will do the class design as suggested above.